### PR TITLE
chore: refactor relative path to better absolute

### LINKED ISSUE #2845: bug: incorrect import paths pointing outside the repository
As of 01050138c66206e48d9d5472e65b6868058a57d0:
```
$ find nwaku/ -mindepth 2 -maxdepth 2 -type f -name '*.nim' -print0 | xargs -0 grep -Fn ../.. | wc -l
93
$ find nwaku/ -mindepth 3 -maxdepth 3 -type f -name '*.nim' -print0 | xargs -0 grep -Fn ../../.. | wc -l
189
$ find nwaku/ -mindepth 4 -maxdepth 4 -type f -name '*.nim' -print0 | xargs -0 grep -Fn ../../../.. | wc -l
5
```

These are all nonsense import paths which Nim 1.6 happens to tolerate, but Nim 2.0 does not.

A sample of `mindepth 2`s:
```
nwaku/tests/test_peer_manager.nim:25:  ../../waku/waku_enr/capabilities,
nwaku/examples/lightpush_publisher.nim:8:  ../../../waku/waku_core,
nwaku/tests/test_relay_peer_exchange.nim:14:  ../../waku/waku_core, ../../waku/waku_node, ./testlib/wakucore, ./testlib/wakunode
nwaku/tests/test_waku_netconfig.nim:7:include ../../waku/node/config
nwaku/tests/test_peer_manager.nim:27:  ../../waku/waku_filter_v2/common,
nwaku/tests/test_waku_metadata.nim:20:  ../../waku/discovery/waku_discv5,
nwaku/tests/test_peer_store_extended.nim:13:  ../../waku/node/peer_manager/waku_peer_store,
nwaku/tests/test_peer_manager.nim:29:  ../../waku/waku_lightpush/common,
nwaku/examples/subscriber.nim:13:  ../../../waku/common/logging,
nwaku/examples/filter_subscriber.nim:9:  ../../../waku/waku_filter_v2/client
nwaku/examples/lightpush_publisher.nim:9:  ../../../waku/waku_lightpush/client
nwaku/tests/test_peer_storage.nim:7:  ../../waku/node/peer_manager/peer_store/waku_peer_storage,
nwaku/tests/test_wakunode_lightpush.nim:6:  ../../waku/waku_lightpush/common,
nwaku/tests/test_waku_protobufs.nim:6:  ../../waku/waku_metadata/rpc,
nwaku/tests/test_waku_keystore_keyfile.nim:4:import ../../waku/waku_keystore, ./testlib/common
nwaku/tests/test_peer_manager.nim:23:  ../../waku/waku_node,
nwaku/tests/test_wakunode.nim:24:  ../../waku/waku_peer_exchange,
nwaku/tests/test_waku_noise_sessions.nim:5:  ../../waku/common/protobuf,
nwaku/tests/test_waku_noise_sessions.nim:6:  ../../waku/utils/noise as waku_message_utils,
nwaku/tests/test_waku_noise.nim:17:  ../../waku/waku_core,
nwaku/examples/subscriber.nim:15:  ../../../waku/waku_core,
nwaku/tests/test_waku_metadata.nim:19:  ../../waku/node/peer_manager,
nwaku/tests/test_wakunode_lightpush.nim:7:  ../../waku/node/peer_manager,
nwaku/library/libwaku.nim:14:  ../../waku/waku_core/topics/pubsub_topic,
nwaku/tests/test_peer_manager.nim:26:  ../../waku/waku_relay/protocol,
```

A sample of the `mindepth 3`s:
```
nwaku/tests/testlib/wakunode.nim:18:  ../../../waku/factory/internal_config,
nwaku/tests/waku_store/test_rpc_codec.nim:8:  ../../../waku/waku_store/common,
nwaku/tests/waku_archive/test_driver_queue_query.nim:9:  ../../../waku/waku_core/message/digest,
nwaku/tests/waku_filter_v2/test_waku_filter_protocol.nim:14:  ../../../waku/waku_core,
nwaku/tests/waku_rln_relay/test_rln_group_manager_onchain.nim:20:  ../../../waku/waku_rln_relay/protocol_types,
nwaku/apps/chat2bridge/chat2bridge.nim:245:  import ../../../waku/common/utils/nat, ../../waku/waku_api/message_cache
nwaku/tests/node/test_wakunode_lightpush.nim:14:  ../../../waku/[
nwaku/tests/wakunode_rest/test_rest_store.nim:26:  ../../../waku/waku_archive,
nwaku/tests/node/test_wakunode_peer_manager.nim:16:  ../../../waku/[
nwaku/tests/waku_archive/test_driver_queue.nim:7:  ../../../waku/waku_archive/driver/queue_driver/index,
nwaku/tests/waku_relay/test_message_id.nim:7:import ../../../waku/waku_relay/message_id, ../testlib/sequtils
nwaku/tests/waku_archive/test_driver_sqlite.nim:5:  ../../../waku/common/databases/db_sqlite,
nwaku/tests/waku_store/test_wakunode_store.nim:22:  ../../../waku/waku_archive,
nwaku/tests/waku_archive/test_waku_archive.nim:15:  ../../../waku/waku_archive/driver/sqlite_driver,
nwaku/tests/waku_rln_relay/test_waku_rln_relay.nim:15:  ../../../waku/waku_rln_relay/rln,
nwaku/tests/waku_store_legacy/test_wakunode_store.nim:26:  ../../../waku/waku_store_legacy,
nwaku/tests/waku_archive/test_driver_queue_pagination.nim:9:  ../../../waku/waku_core,
nwaku/tests/waku_archive/test_partition_manager.nim:5:  ../../../waku/waku_archive/driver/postgres_driver/partitions_manager,
nwaku/tests/waku_archive/test_waku_archive.nim:11:  ../../../waku/common/databases/db_sqlite,
nwaku/tests/waku_archive/test_retention_policy.nim:12:  ../../../waku/waku_archive/retention_policy/retention_policy_size,
nwaku/tests/waku_rln_relay/test_waku_rln_relay.nim:13:  ../../../waku/waku_core,
nwaku/tests/wakunode_rest/test_rest_store.nim:19:  ../../../waku/node/peer_manager,
nwaku/apps/liteprotocoltester/filter_subscriber.nim:16:  ../../../waku/common/logging,
nwaku/tests/waku_store/test_client.nim:6:  ../../../waku/
nwaku/apps/liteprotocoltester/lightpush_publisher.nim:13:  ../../../waku/waku_core,
```

And the `mindepth 4`s:
```
nwaku/tests/waku_rln_relay/rln/test_rln_interface.nim:3:import ../../../../waku/waku_rln_relay/rln/rln_interface, ./buffer_utils
nwaku/tests/waku_rln_relay/rln/buffer_utils.nim:1:import ../../../../waku/waku_rln_relay/rln/rln_interface
nwaku/tests/node/peer_manager/test_peer_manager.nim:12:  ../../../../waku/node/peer_manager/peer_manager
nwaku/tests/waku_core/topics/test_pubsub_topic.nim:5:import ../../../../waku/[waku_core/topics/pubsub_topic], ../../testlib/[wakucore]
nwaku/tests/waku_core/topics/test_sharding.nim:3:import ../../../../waku/waku_core/topics, ../../testlib/[wakucore, tables, testutils]
```

### PR DESCRIPTION
**Pull request by darshankabariya**
closes #2845

### CODE REVIEW

**Review by gabrielmer — APPROVED** [review]
LGTM, thanks so much!

File: apps/chat2bridge/nim.cfg
Comment (NagyZoltanPeter) [inline apps/chat2bridge/nim.cfg:4]: Me, personally would prefer better to always use the `waku/` prefix all the time we refer the library from an app.
For me it is harder to read from where are those imports coming.
WDYT?

**Review by NagyZoltanPeter — APPROVED** [review]
In general I like the approach, but I would better leave the reference to the `waku/` directory and just eliminate the `../..` like stuff.
I think the would help reading the imports. Similarly we have chronos/... nim-eth/, etc references in imports.
So treat waku as a lib for tests and apps. WDYT?

**Comment by NagyZoltanPeter** [comment]
@darshankabariya: Please consider to wait with merging it with in sync with next release, maybe just before cut off and release-notes. That will help other's branch rebasing before merge, otherwise there can be many conflicts for some branches.
cc: WDYT @gabrielmer, @Ivansete-status

**Comment by darshankabariya** [comment]
> In general I like the approach, but I would better leave the reference to the `waku/` directory and just eliminate the `../..` like stuff. I think the would help reading the imports. Similarly we have chronos/... nim-eth/, etc references in imports. So treat waku as a lib for tests and apps. WDYT?

Got it ! it's better clarity then this.

**Comment by darshankabariya** [comment]
> @darshankabariya: Please consider to wait with merging it with in sync with next release, maybe just before cut off and release-notes. That will help other's branch rebasing before merge, otherwise there can be many conflicts for some branches. cc: WDYT @gabrielmer, @Ivansete-status

Thank you for the suggestion. I understand the concern regarding potential conflicts. I'll hold off on merging my changes and will plan accordingly.

**Review by Ivansete-status — COMMENTED** [review]
It is looking beautiful, thanks so much for it! :raised_hands:

Notice that we need to perform some minor adjustments to the tests so that the Ubuntu tests pass as well :P

I agree with you @NagyZoltanPeter that we need to wait until the `v0.30.0` is released. @darshankabariya , you will need to rebase from master after the `v0.30.0` is released.

On the other hand, we still need to push it a bit further as there are some pending files to be reviewed.

We need the following commands to return `0` (maybe we can add them as a "lint" check on every PR:)

```
find nwaku/ -mindepth 2 -maxdepth 2 -type f -name '*.nim' -print0 | xargs -0 grep -Fn ../.. | wc -l
find nwaku/ -mindepth 3 -maxdepth 3 -type f -name '*.nim' -print0 | xargs -0 grep -Fn ../../.. | wc -l
find nwaku/ -mindepth 4 -maxdepth 4 -type f -name '*.nim' -print0 | xargs -0 grep -Fn ../../../.. | wc -l
```

File: tests/node/test_wakunode_relay_rln.nim
Comment (darshankabariya) [reply tests/node/test_wakunode_relay_rln.nim:15]: Actually, I used a script for these refactor changes. It's open bracket, but I will bundle everything together where necessary and complete the remaining refactoring as well. Thanks for review.

File: tests/node/test_wakunode_lightpush.nim
Comment (darshankabariya) [reply tests/node/test_wakunode_lightpush.nim:99]: i am using that version only. If you want the original state, I will disable it for a while to revert to the previous state.

Comment (darshankabariya) [reply tests/node/test_wakunode_lightpush.nim:99]: image

File: tests/node/test_wakunode_relay_rln.nim
Comment (darshankabariya) [reply tests/node/test_wakunode_relay_rln.nim]: Hi @Ivansete-status,
Please review the PR one more time. There have been many changes, and the build has failed multiple times, which is why it’s taking a bit longer than expected. Regarding `test-ubuntu-latest`, I believe the issue is not related to these changes. Any comments on the PR are welcome.

**Comment by Ivansete-status** [comment]
Thanks so much for that @darshankabariya ! :heart: 
I just merged it because I needed it in those changes for "bump dependencies", where we will start using Nim 2.0.8 ( if everything goes well of course ;P )

**Pull request merged**
